gpflow/misc.py

```diff
@@ -153,13 +153,13 @@ def int_shape(x):
     # Check M is a valid triangle number
     assert((matrix * (N + 1)) == (2 * M))
     """
-    indices = list(zip(*np.tril_indices(N)))
-    indices = tf.constant([list(i) for i in indices], dtype=tf.int64)
+    D = int(vectors.shape[0])
```

Comment:
This will break if vectors has unknown shape. Maybe this is fine for most uses, but I think the motivation for the existing code is to deal with dynamic shapes. If this code is adopted then D should probably also be an input.